vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -88,13 +96,19 @@ type internal ProjectInfoManager
         checkerProvider: FSharpCheckerProvider,
         [<Import(typeof<SVsServiceProvider>)>] serviceProvider: System.IServiceProvider
     ) =
-    // A table of information about projects, excluding single-file projects.  
+
+    // A table of information about projects, excluding single-file projects.
     let projectTable = ConcurrentDictionary<ProjectId, Refreshable<ProjectId[] * FSharpProjectOptions>>()
 
     // A table of information about single-file projects.  Currently we only need the load time of each such file, plus
     // the original options for editing
     let singleFileProjectTable = ConcurrentDictionary<ProjectId, DateTime * FSharpProjectOptions>()
 
+    // Accumulate sorces and references for each project file
+    let lockObject = new Object()
+    let sourceFiles = new Dictionary<string, string[]>()
```

Comment:
I'm pretty sure you can just use ConcurrentDictionary for these given the simple pattern of access.  That's consistent with the two uses of ConcurentDictionary above.